### PR TITLE
New design changes for Filtering Notifications

### PR DESCRIPTION
**Pull request by sharvaniharan**
**Phab:** https://phabricator.wikimedia.org/T288068

### CODE REVIEW

**Comment by cooltey** [comment]
May I know the reason why there are two PRs related to [this ticket](https://phabricator.wikimedia.org/T288068)?

**Comment by sharvaniharan** [comment]
@cooltey The designs were updated on Friday... it was a significant change and i did not want to flip the old PR back to WIP .. it would be confusing... thats why opened a new one.

**Review by cooltey — CHANGES_REQUESTED** [review]
Found two following issues:

1. In a fresh install and first-time enter the notification page, the list will be empty.



2. The "Japanese" and "Traditional Chinese" items are not clickable. Also, should the "all" types of the button is able to be check/uncheck all?

**Comment by sharvaniharan** [comment]
Thanks for catching this @cooltey i have fixed the bugs, will ask Robin about the toggle of "all wikis" feel free to merge if everything else checks out and i can just make a tiny tweak PR for the toggle.

File: app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt [hunk 1]
```diff
@@ -495,24 +510,45 @@ class NotificationActivity : BaseActivity() {
...
             }
 
             FeedbackUtil.setButtonLongPressToast(notificationFilterButton)
         }
+
+        private fun updateFilterIconAndCount() {
+            val fullWikiAndTypeListSize = NotificationsFilterActivity.allWikisList().size + NotificationsFilterActivity.allTypesIdList().size
+            val delimitedFiltersSizeString = Prefs.notificationsFilterLanguageCodes.orEmpty().split(",").filter { it.isNotEmpty() }.size
+            val enabledFilters = fullWikiAndTypeListSize - delimitedFiltersSizeString
+            if (enabledFilters == 0 || Prefs.notificationsFilterLanguageCodes == null) {
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt:543]: Can we use `.isNullOrEmpty()` on the `Prefs.notificationsFilterLanguageCodes`?

File: app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt [hunk 2]
```diff
@@ -266,24 +273,32 @@ class NotificationActivity : BaseActivity() {
             if (!currentSearchQuery.isNullOrEmpty() && n.contents != null && !n.contents.header.contains(currentSearchQuery!!)) {
                 continue
             }
-            notificationContainerList.add(NotificationListItemContainer(n))
+            val filterList = mutableListOf<String>()
+            filterList.addAll(StringUtil.csvToList(Prefs.notificationsFilterLanguageCodes.orEmpty()).filter { NotificationCategory.isFiltersGroup(it) })
+            if (filterList.contains(n.category) || Prefs.notificationsFilterLanguageCodes == null) notificationContainerList.add(NotificationListItemContainer(n))
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt:278]: Can we use `.isNullOrEmpty()` on the `Prefs.notificationsFilterLanguageCodes`?

File: app/src/main/java/org/wikipedia/notifications/NotificationsFilterActivity.kt [hunk 3]
```diff
@@ -31,92 +31,137 @@ class NotificationsFilterActivity : BaseActivity() {
...
+        fun bindItem(filterHeader: String) {
+            headerText.text = filterHeader
         }
     }
 
-    class NotificationsFilterAdapter(val context: Context) :
-        RecyclerView.Adapter<NotificationFilterItemHolder>(), NotificationFilterItemView.Callback {
+    class NotificationsFilterAdapter(val context: Context, private val filtersList: MutableList<Any>) :
+        RecyclerView.Adapter<DefaultViewHolder<*>>(), NotificationFilterItemView.Callback {
         var app: WikipediaApp = WikipediaApp.getInstance()
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/notifications/NotificationsFilterActivity.kt:72]: Redundant variable.

File: app/src/main/java/org/wikipedia/notifications/NotificationFilterItemView.kt [hunk 4]
```diff
@@ -23,39 +28,73 @@ class NotificationFilterItemView constructor(context: Context, attrs: AttributeS
...
-            binding.notificationFilterLanguageCode.text = langCode
+    fun setContents(filter: Filter) {
+        this.filter = filter
+        binding.notificationFilterTitle.text = getTitleFor(filter.languageCode)
+        binding.notificationFilterCheck.visibility = if (filter.isEnabled()) View.VISIBLE else View.GONE
+        getTitleCodeFor(filter.languageCode)?.let {
+            binding.notificationFilterLanguageCode.text = filter.languageCode
             binding.notificationFilterLanguageCode.visibility = View.VISIBLE
-            ViewUtil.formatLangButton(binding.notificationFilterLanguageCode, langCode, SearchFragment.LANG_BUTTON_TEXT_SIZE_SMALLER, SearchFragment.LANG_BUTTON_TEXT_SIZE_LARGER)
+            ViewUtil.formatLangButton(binding.notificationFilterLanguageCode, it, SearchFragment.LANG_BUTTON_TEXT_SIZE_SMALLER, SearchFragment.LANG_BUTTON_TEXT_SIZE_LARGER)
```
Comment (cooltey) [inline app/src/main/java/org/wikipedia/notifications/NotificationFilterItemView.kt:62]: Looks like the language code is not correct. Should it be `langCode`?

Comment (cooltey) [reply app/src/main/java/org/wikipedia/notifications/NotificationFilterItemView.kt:62]: I also notice the text color is incorrect in dark mode, would you update that with a proper attritube color?

**Review by cooltey — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt (hunk 1 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt:543]: This doesnt apply when empty @cooltey

File: app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt (hunk 2 above)
Comment (sharvaniharan) [reply app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt:278]: Same as above

**Comment by sharvaniharan** [comment]
Done @cooltey

**Pull request merged**